### PR TITLE
Feature/mj season statistics add winningest method

### PR DESCRIPTION
**Pull request by MWMJohnson**
## High level what you did: 
Added Test and Methods for both winningest and worst coach on League Statistics class and spec files.
Created Helper method called coaches_records
### Details:
#coaches_records, returns a hash with a coach as the key, and the value is their record of wins, losses, and ties.
#winningest_coach returns a string with the name of the coach with the highest win percentage.
#worst_coach returns a string with the name of the coach with the highest loss percentage.
## Issues: YES
Having issues writing my test for my helper method coaches_record, my expected statement if not working. Do I have a typo or do I need to use a stub rather? If so not sure how to proceed.

## Questions:
Thoughts on the above?
## Overall:
- [x] Tests are written for all methods added
- [ ] Rspec running clean
- [x] Bugs, need help, etc.

### CODE REVIEW

**Comment by aplasket** [comment]
hey @MWMJohnson I think these methods need to pass in an argument of each season. Based on the data in the spec_harness, it looks like we're supposed to calculate the winningest and worse coach per season, and the most and least accurate team per season, not overall.

File: spec/season_statistics_spec.rb [hunk 1]
```diff
@@ -23,19 +23,32 @@
...
+  describe "#coach_record" do 
+  it "can list the loss, ties, and wins of each coach" do 
+    expected = {"Adam Oates" => {:games_coached=>183, :losses=>79, :ties=>44, :wins=>60},
+                "Alain Vigneault" => {:games_coached=>523, :losses=>192, :ties=>100, :wins=>231},
+                "Barry Trotz" => {:games_coached=>521, :losses=>179, :ties=>99, :wins=>243},
+                "Bill Peters" => {:games_coached=>328, :losses=>141, :ties=>62, :wins=>125},
+                "Bob Boughner" => {:games_coached=>82, :losses=>28, :ties=>16, :wins=>38},
+                "Bob Hartley" => {:games_coached=>305, :losses=>148, :ties=>52, :wins=>105},
+                "Bruce Boudreau" => {:games_coached=>511, :losses=>161, :ties=>108, :wins=>242}}
+    # expect(@season_stats.coach_record).to eq(expected)
```
Comment (aplasket) [inline spec/season_statistics_spec.rb]: it looks like this "expected" variable is not holding all coaches that are printing in the method. When I stuck a pry in on ln 36 of season_stats and printed "coaches" it printed a super long list of coaches and their data. So I think the test is working for sure, but the test here is just not including everyone. 

I'd suggest making an assertion that the method results output a hash, and that a specific key is equal to a specific value. For example:

expect(@season_stats.coach_record).to be_a(Hash)

expect(@season_sats.coach_record["John Tortorella"].to eq({:games_coached=>392,

File: spec/season_statistics_spec.rb [hunk 2]
```diff
@@ -23,19 +23,32 @@
...
+  end
+end
+  
+  describe "#winningest_coach" do
+    it " names the Coach with the best win percentage for the season" do 
+      expect(@season_stats.winningest_coach).to be_a(String)
+      expect(@season_stats.winningest_coach).to eq("Dan Lacroix")
+    end
```
Comment (aplasket) [inline spec/season_statistics_spec.rb:72]: in the spec_harness it looks like we're supposed to pass in an argument of season id and calculate the winningest coach for each season. 

here's the assertions from the spec harness for this test:
it "#winningest_coach" do
    expect(@stat_tracker.winningest_coach("20132014")).to eq "Claude Julien"
    expect(@stat_tracker.winningest_coach("20142015")).to eq "Alain Vigneault"
  end

File: spec/season_statistics_spec.rb [hunk 3]
```diff
@@ -23,19 +23,32 @@
...
+  end
 
-#       # return value string
-#     end
-#   end
+  describe "#worst_coach" do
+    it " names the Coach with the worst win percentage for the season" do 
+      expect(@season_stats.worst_coach).to be_a(String)
+      expect(@season_stats.worst_coach).to eq("Martin Raymond")
+    end
```
Comment (aplasket) [inline spec/season_statistics_spec.rb:120]: hey @MWMJohnson same with the winningest coach, i think we're supposed to pass in an argument of the season id here, and print the worst coach per season

based on the spec_harness this is what the test is expecting the output values are:
 it "#worst_coach" do
    expect(@stat_tracker.worst_coach("20132014")).to eq "Peter Laviolette"
    expect(@stat_tracker.worst_coach("20142015")).to eq("Craig MacTavish").or(eq("Ted Nolan"))
  end

File: spec/season_statistics_spec.rb (hunk 1 above)
Comment (MWMJohnson) [reply spec/season_statistics_spec.rb]: You are absolutely correct, was having a brainfart, I have edited this test.

File: spec/season_statistics_spec.rb (hunk 2 above)
Comment (MWMJohnson) [reply spec/season_statistics_spec.rb:72]: Fixed!

File: spec/season_statistics_spec.rb (hunk 3 above)
Comment (MWMJohnson) [reply spec/season_statistics_spec.rb:120]: Fixed!

**Comment by MWMJohnson** [comment]
All tests and methods are working as described, except there appears to be an error in spec harness, for worst coach for season "20142015", should be Dallas Eakins.

Let met know if you have any questions.

**Comment by goldenll** [comment]
Great job, looks like you put lots of work to get the expected results. Good work!

**Pull request merged**